app/controllers/events_controller.rb

```diff
@@ -114,7 +114,8 @@ def index
     dois = total.positive? && params[:extra] ? facet_by_dois(response.response.aggregations.dois.buckets) : nil
     dois_usage = total.positive? && params[:extra] ? facet_by_dois(response.response.aggregations.dois_usage.dois.buckets) : nil
     dois_citations = total.positive? && params[:extra] ? facet_citations_by_year(response.response.aggregations.dois_citations) : nil
-
+    # unique_citations = total.positive? && params[:extra] ? facet_citations_by_dois(response.response.aggregations.unique_citations.dois.buckets) : nil
```

Comment:
Why is this commented out? And what is the relationship between `dois_citations` and `unique_citations`, and do you need both?

Reply:
Because this aggregation will fail unless the citation_id is already in the index-mapping. So the plan is to add citation_id to the mapping and then enable all the aggregations in a secondary step.  

Do we need `dois_citations` and `unique_citations`? unfortunately kind of yes. The `unique_citations` aggregation is very fast at getting us the total number of unique citations. Unfortunately is a numeric aggregation, hence, we cannot have further sub aggregation; e.g., by year for the chart. That is why we are using `dois_citations` at the moment. 

There seems to be still discussion (robins document) of what criteria will be used to remove duplicated in the chart. But for the moment I will deal with that filtering somewhere else.

Reply:
What sub-aggregations do you need? I thought we said that for aggregations by year, `unique_citations` should give the same numbers.

Aggregations are fast, but we should still be careful that we only do the aggregations we really need.

Reply:
> What sub-aggregations do you need? I thought we said that for aggregations by year, unique_citations should give the same numbers.

I need a sub-aggregation on the unique items aggregation. 

AS mentioned in the GitHub issue https://github.com/datacite/lupo/issues/232 we obtain the non-duplicated aggregation with the cardinality aggregation metric. Unfortunately, this aggregation doesn't support sub-aggregations. Therefore we cannot get a subsequent aggregation with the time distribution for the chart. So if there are duplicated the chart data would differ from the citations count.

A solution I am working on is to use `collapsing fields`. Collapsing fields filter the list of results but do not affect the aggregations (with ES you cannot have everything :) hehe). One idea using collapsing fields is as follow.  

1. We call API with the cardinality aggregation and the collapsing field filter.
1. We estimate the number of duplicates with the cardinality aggregation. (in the Search Client)
1. If there are duplicates we generate a new yearly citation distribution data-array based on the collapsed results list from the API. This gives us a dataset without duplicates.
1. otherwise we use the default aggregation by year that the API already gives us.

Reply:
https://github.com/datacite/lupo/pull/306/commits/d454e0d32e8a73319bdff7f69f1db2a5fc1fddbc contains the collapsing fields filter